src/runtime/hexagon/hexagon_device_api.h

```diff
@@ -153,6 +162,8 @@ class HexagonDeviceAPI final : public DeviceAPI {
     return runtime_threads ? runtime_threads.get() : nullptr;
   }
 
+  HexagonUserDMA* UserDMA() { return runtime_dma ? runtime_dma.get() : nullptr; }
```

Comment:
Either we need to `CHECK` that runtime_dma is not null here or in the calling scope.  We are doing neither.  All callers of `UserDMA()` immediately dereference it.  My thought would be to `CHECK` non null here.

You might apply this change to the `ThreadManager()` - I believe it has the same issue.

Reply:
I thought about that, but then thought std::make_unique would throw or assert.  Let me dig into the behavior.  If it doesn't throw or assert, I'll add checks for all three managers.

Reply:
I think the `CHECK` that I am recommending here is really a check that the user has called `AcquireResources` which calls `make_unique` before calling `UserDMA`.  If indeed `make_unique` cannot return a nullptr then we are all good.  If it can return nullptr, then you should also `CHECK` after `make_unique` in `AcquireResources`